### DISCUSSION TITLE
Tasmota rule - controls underfloor heating (8 loops) + boiler + ESP32-8ch relay board + LYWSD03MMC

### DISCUSSION THREAD

**Discussion by ON-state** [arendst/Tasmota#22449] [Q&A]
Hey,

I have built a project to control underfloor heating using 8 actuators based on ESP32+8CH relay board, 4x Xiaomi LYWSD03MMC temperature sensors , boiler is controled by re-build SonOff O/I [192.168.1.100] and Tasmota

I am developing a rule based on the logic:
- turn on the furnace if any of the LYWSD03MMC is below the set temperature
- turn off the furnace if any of all LYWSD03MMC are above or equal to the set temperature
- if the furnace is already on and any of the LYWSD03MMC is below the set temperature, do not re-send the message turn on the furnace

I have an idea for a rule but it requires adding the logic function to Tasmota

With such a complicated rule, is adding the logic the only option or can I write it based on IF ELSE or ON ENDONE?

At the moment I have created a rule where only one of the sensors controls the shutdown of the furnace, but this is not an ideal solution because if the condition is met for it, it turns off the furnace and the loop assigned to it, but the rest of the loops where the sensors do not meet the condition are still open.

`Rule1 
ON ATCae8e6c#Temperature DO var1 %value% ENDON  
ON LYWSD035f8c22#Temperature DO var2 %value% ENDON  
ON LYWSD03a8f47c#Temperature DO var3 %value% ENDON  
ON LYWSD037a6e7c#Temperature DO var4 %value% ENDON 

ON var1#state<21 DO BACKLOG Power6 1; Power7 1;  Power8 1; Power9 1 ENDON  
ON var1#state>=21 DO BACKLOG Power6 0; Power7 0; Power8 0; Power9 0 ENDON  
ON var2#state<22 DO Power3 1 ENDON 
ON var2#state>=22 DO Power3 0 ENDON  
ON var3#state<20.5 DO Power4 1 ENDON 
ON var3#state>=20.5 DO Power4 0 ENDON  
ON var4#state<21 DO Power5 1 ENDON 
ON var4#state>=21 DO Power5 0 ENDON  

ON var2#state<22 DO BACKLOG IF (var5==0); Websend [192.168.1.100] Power1 1; var5 1;Power1 1 ENDIF ENDON 
ON var2#state>=22 DO BACKLOG IF (var5==1); Websend [192.168.1.100] Power1 0; var5 0;Power1 0;Power3 0;Power4 0;Power5 0;Power6 0;Power7 0;Power8 0;Power9 0 ENDIF ENDON`

**Comment by Noschvie** [comment]
Have a look to Berry, it’s supported by the ESP32.
